### PR TITLE
Move RobotInformation to mbf_utility, as it can be used generally

### PR DESCRIPTION
**Pull request by corot**
I also fixed some clang-tidy warnings in the process

### CODE REVIEW

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h [hunk 1]
```diff
@@ -89,7 +87,7 @@ namespace mbf_abstract_nav
      * @param tf_listener_ptr Shared pointer to a common tf listener
      */
     AbstractControllerExecution(
-        const std::string name,
```
Comment (Bitfroest) [inline mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h:92]: format of `const std::string& name`

Note that parameter 'name' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions clang-tidy

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_execution_base.h [hunk 2]
```diff
@@ -39,11 +39,12 @@
 #ifndef MBF_ABSTRACT_NAV__ABSTRACT_EXECUTION_BASE_H_
 #define MBF_ABSTRACT_NAV__ABSTRACT_EXECUTION_BASE_H_
 
-#include <mbf_abstract_nav/MoveBaseFlexConfig.h>
 #include <boost/thread.hpp>
 #include <boost/chrono/duration.hpp>
 #include <boost/chrono/thread_clock.hpp>
 
+#include <mbf_abstract_nav/MoveBaseFlexConfig.h>
```
Comment (Bitfroest) [inline mbf_abstract_nav/include/mbf_abstract_nav/abstract_execution_base.h:46]: `#include "mbf_abstract_nav/MoveBaseFlexConfig.h"`

prefers the workspace version more than the apt package

File: mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h (hunk 1 above)
Comment (Bitfroest) [reply mbf_abstract_nav/include/mbf_abstract_nav/abstract_controller_execution.h:92]: Seems to be different throughout files, do you use a clang-format or something?

File: mbf_abstract_nav/include/mbf_abstract_nav/recovery_action.h [hunk 3]
```diff
@@ -41,22 +41,25 @@
...
 
-#include "mbf_abstract_nav/abstract_action_base.hpp"
-#include "mbf_abstract_nav/abstract_recovery_execution.h"
-#include "mbf_abstract_nav/robot_information.h"
+#include <boost/thread/condition_variable.hpp>
 #include <actionlib/server/action_server.h>
+
 #include <mbf_msgs/RecoveryAction.h>
-#include <boost/thread/condition_variable.hpp>
+#include <mbf_utility/robot_information.h>
```
Comment (Bitfroest) [inline mbf_abstract_nav/include/mbf_abstract_nav/recovery_action.h:48]: `#include "mbf_utility/robot_information.h"`

**Review by spuetz — CHANGES_REQUESTED** [review]
Can you separate the style changes from the logic changes. Meaning two PRs.

**Comment by corot** [comment]
> Can you separate the style changes from the logic changes. Meaning two PRs.

Done

**Comment by spuetz** [comment]
LGTM

**Pull request merged**